magento/pwa-studio
# PWA-1790: Spike: Page Builder support - How to enable components to m…

## Description

When scroll bar is configured in system to "always show" - it's width is not considered in full viewport width (100vw), so added calculation to subtract it from viewport width, also `ResizeObserver` is used to track if scrollbar appeared after page height was increased.


## Related Issue







Closes #[PWA-1790](https://jira.corp.magento.com/browse/PWA-1790).

## Acceptance





### Verification Stakeholders



### Specification




### Verification Steps




1. Add 3 rows using PageBuilder to the CMS page.
2. Configure each row to have different backgrounds and appearance settings:
   - Contained
   - Full width
   - Full bleed
3. See rows content types looks correct on the Venia storefront:
    - Full Bleed and Full Width rows background should take full width of the viewport.
    - Full Bleed row inner content should take full width of the viewport
4. Configure system to have a different scroll bar behaviour
    - (on MacOS - System Preferences -> General -> Show scroll bars: Automatically/Always)
5. See how scroll bars influence on rows appearance in the viewport (no horizontal scrollbar should appear)
6. Check other possible row content type settings: Paddings, Margins, Video Background, Parallax, etc. 
7. Run MFTF
8. Verify new/impacted UI elements in supported browsers (chrome, firefox, safari) and devices (Safari- iOS, Chrome- Android)


## Screenshots / Screen Captures (if appropriate)
![image](https://user-images.githubusercontent.com/8600171/122238540-d38ae700-ce85-11eb-8ed2-32615b49fe21.png)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
